realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -237,6 +240,16 @@ public RxObservableFactory getRxFactory() {
         return rxObservableFactory;
     }
 
+    /**
+     * Returns whether this Realm is read only or not. Read-only Realms cannot be modified and will throw an
+     * {@link IllegalStateException} if {@link Realm#beginTransaction()} is called on it.
+     *
+     * @return {@code true} if this Realm is read only, {@code false} if not.
+     */
+    public boolean isReadOnly() {
+        return readOnly;
+    }
+
     @Override
     public boolean equals(Object obj) {
```

Comment:
No, the schema cache should detect different readOnly modes and throw.  It is a bug it wasn't added to `equals()`.

We can never detect this across processes though.